test/creditguild-test.js

```diff
@@ -0,0 +1,206 @@
+      const initialVouchAmount = await creditGuild.vouchAmount();
+      expect(VOUCHAMOUNT).to.equal(initialVouchAmount);
+
+      // happy path - test changing the vouch amount with the setVouchAmount function
+      const newVouchAmount = ethers.utils.parseEther("50");
+      await creditGuild.setVouchAmount(newVouchAmount);
+      const vouchAmount = await creditGuild.vouchAmount();
+      expect(vouchAmount).to.equal(newVouchAmount);
+
+    });
+  });
+
+  describe("CheckIsMember", function () {
+    it("Should check if address is a member", async function () {
+
+      // MEMBER1 should be a member
+      const potentialMember = MEMBER1.address;
+      const tx = await creditGuild.checkIsMember(potentialMember);
+      const resp = await tx.wait();
+      result = resp.events[resp.events.length - 1].args.result;
+      expect(result).to.equal(true);
+
+      // JACK should not be a member
+      const potentialMember2 = JACK.address;
+      const tx2 = await creditGuild.checkIsMember(potentialMember2);
+      const resp2 = await tx2.wait();
+      result2 = resp2.events[resp2.events.length - 1].args.result;
+      expect(result2).to.equal(false);
+
+    });
+  });
+
+  describe("Register", function () {
+    it("Should register an array of members", async function () {
+
+      // unhappy path - should revert because ALICE is not a member
+      await expect(creditGuild.register(
+        [MEMBER1.address, MEMBER2.address, ALICE.address]
+      )).to.be.revertedWith('!member');
+
+      // unhappy path - all 3 of the members are not vouching for msg.sender
+      // TODO: with the mock set as always vouching 1, we are unable to test this
+
+      // happy path
+      const addressToRegister = ALICE.address;
+
+      const tx = await creditGuild.checkIsMember(addressToRegister);
+      const resp = await tx.wait();
+      result = resp.events[resp.events.length - 1].args.result;
+      expect(result).to.equal(false);
+
+      await creditGuild.connect(ALICE).register([MEMBER1.address, MEMBER2.address, MEMBER3.address]);
+
+      const tx1 = await creditGuild.checkIsMember(addressToRegister);
+      const resp1 = await tx1.wait();
+      result1 = resp1.events[resp1.events.length - 1].args.result;
+      expect(result1).to.equal(true);
+
+      // unhappy path - should revert because msg.sender is already a member
+      await expect(creditGuild.connect(ALICE).register(
+        [MEMBER1.address, MEMBER2.address, MEMBER3.address]
+      )).to.be.revertedWith('member');;
+
+    });
+  });
+
+  describe("Claim Vouch", function () {
+    it("Should allow an address to claim a vouch", async function () {
+
+      // get current vouch amount
+      vouchAmount = await creditGuild.vouchAmount();
+
+      tx = await creditGuild.claimVouch();
+      const resp = await tx.wait();
+      result = resp.events[resp.events.length - 1].args.vouchAmount;
+      expect(result).to.equal(vouchAmount);
+
```

Comment:
Here you will also need to check the UserManager... in fact there is not much point testing this in a unit test.